pkgs/tools/backup/restic/rest-server.nix

```diff
@@ -1,22 +1,27 @@
-{ lib, buildGoModule, fetchFromGitHub }:
+{ lib, buildGoModule, fetchFromGitHub, fetchpatch }:
 
 buildGoModule rec {
   pname = "restic-rest-server";
   version = "0.11.0";
+  owner = "restic";
+  repo = "rest-server";
 
   src = fetchFromGitHub {
-    owner = "restic";
-    repo = "rest-server";
+    inherit owner repo;
     rev = "v${version}";
     hash = "sha256-ninPODztNzvB2js9cuNAuExQLK/OGOu80ZNW0BPrdds=";
   };
 
   vendorSha256 = "sha256-8x5qYvIX/C5BaewrTNVbIIadL+7XegbRUZiEDWmJM+c=";
 
-  preCheck = ''
-    substituteInPlace cmd/rest-server/main_test.go \
-      --replace "/tmp/restic" "/build/restic"
-  '';
+  patches = [
+    (fetchpatch {
+      name = "backport_rest-server_tests_os.TempDir.patch";
+      url = "https://github.com/${owner}/${repo}/commit/a87a50ad114bdaddc895413396438df6ea0affbb.patch";
```

Comment:
Please stop interpolating every possible variable. If the org or the repo is renamed GitHub has a redirection and if it is replaced we still have FOD in the hydra cache.

Reply:
Please be mindful of your communication.
I have 2 or 3 contributions to nixpkgs. I'm rather new. It isn't like I'm insisting to not use idiomatic Nix in my PR, so "please stop [...]" prelude is rather uncalled for.

I tried to glean if this is a common pattern or with this repo search:
```
% rg -le 'github\.com/\$\{owner\}'
pkgs/applications/editors/cudatext/update.sh
pkgs/applications/editors/eclipse/plugins.nix
pkgs/applications/video/losslesscut-bin/appimage.nix
pkgs/applications/video/losslesscut-bin/windows.nix
pkgs/applications/video/losslesscut-bin/dmg.nix
pkgs/development/libraries/arguments/default.nix
pkgs/development/libraries/science/biology/bicgl/default.nix
pkgs/development/web/deno/update/librusty_v8.ts
pkgs/development/tools/json2hcl/default.nix
pkgs/development/tools/gomplate/default.nix
pkgs/tools/backup/restic/rest-server.nix
pkgs/tools/networking/dd-agent/datadog-agent.nix
```

I will send another PR, removing interpolation.